### ISSUE TITLE
Why GQA instructions are different from the paper?

### ISSUE THREAD

**Issue by Guitaricet** [ashkamath/mdetr#25]
Hi! Thank you for MDETR, it is an amazing idea and special thanks for documenting your code. It helped me a lot in understanding MDETR.

I have a question about the GQA dataset fine-tuning for visual question answering.

The paper says

> [we] fine-tune first for 5 epochs on the unbalanced all GQA split, followed by 10 epochs on the balanced split similar to what is done in prior work [28, 5]. During the first 5 epochs, we train the modulated detection losses along with the question answering, but put a weight on question answering loss that encourages the model to focus more on this task. For the balanced split fine-tuning, we only use the question answering loss.

However, "reproduce results" instructions in [gqa.md](https://github.com/ashkamath/mdetr/blob/main/.github/gqa.md) suggest fine-tuning for 125 epochs on either "all" or "balanced" split. It also seem to use detection loss for all of the epochs.

Could you clarify why these instructions are different and what kind of the results one can expect from the github instructions vs paper instructions. Currently, after training for 25 epochs with github instructions (balanced dataset) I have about 53% validation accuracy (`gqa_accuracy_answer_total_unscaled`) which is much lower than the number in the paper (62%).

Also, how important have you found the detection objective for this task?

**Comment by ashkamath** [comment]
Hi! Thanks for your interest :)

Sorry for the confusion - we ran experiments and seem to have not correctly updated the paper. We will be uploading a revision very soon. I'll describe our setup below, please let me know if you have any further questions. 

This was the command that we used for first fine-tuning on the "all" split, which is essential to getting good final performance that we report, after further finetuning on the balanced split.

*(make sure to change the configs/gqa.json to point to the all split)*
python run_with_submitit.py --dataset_config configs/gqa.json --ngpus 8 --ema --epochs 125 --epoch_chunks 25 --do_qa --split_qa_heads --lr_drop 150 --load https://zenodo.org/record/4721981/files/pretrained_resnet101_checkpoint.pth--nodes 4 --batch_size 4 --no_aux_loss --qa_loss_coef 25 --lr 1.4e-4 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5

For the finetuning on all, we do use the detection losses and you can see we have --qa_loss_coef 25, which is the weight on the qa loss to encourage the model to focus on this task. Also, unlike regular MDETR training, we have --no_aux_loss which further reduces the emphasis on the detection losses (this means we are not supervising each layer, but only the last layer). We did have ablation runs which did not have the detection losses at all during this phase but these runs performed consistently worse. 

Once we have a model finetuned on the "all" set,  we finetune on the "balanced" set (GQA leaderboard eval script focuses on this split which is why its common practice to do this) for 10 epochs using the command given below:

*(make sure to change the configs/gqa.json to point to the balanced split)*
python run_with_submitit.py --dataset_config configs/gqa.json --ngpus 8 --ema --epochs 10 --do_qa --split_qa_heads  --load /path/to/model/finetuned/on/all--nodes 4 --batch_size 4 --no_aux_loss --qa_loss_coef 25  --ce_loss_coef 0 --bbox_loss_coef 0 --giou_loss_coef 0 --lr 7e-5 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5

We have not uploaded the intermediate checkpoint after finetuning on all, but you can tell me if you need it I can provide it to you. 

Hope this makes sense and I'll update the readme to be more clear as well. 

Best, 
Aish

**Comment by Guitaricet** [comment]
Thank you! This makes sense.
